src/layers/CementLayer.ts

```diff
@@ -56,24 +55,29 @@ export class CementLayer extends WellboreBaseComponentLayer {
   }
 
   createCementShape = (cement: Cement, casings: Casing[], holes: HoleSize[]): CementShape => {
-    const attachedCasing = findCasing(cement.casingId, casings);
-    const bottomOfCement = attachedCasing.end;
+    // Merge deprecated casingId and casingIds array
+    const casingIds = [cement.casingId, ...(cement.casingIds || [])].filter((id) => id);
 
-    const { outerCasings, holes: overlappingHoles } = findIntersectingItems(cement, attachedCasing, casings, holes);
+    const attachedCasings = casingIds.map((casingId) => casings.find((casing) => casing.casingId === casingId));
+    if (attachedCasings.length === 0 || attachedCasings.includes(undefined)) {
```

Comment:
Is it possible to have casings that are undefined? What if they are null?

Reply:
we wouldn't find the potential null casings in the search on previous line.
But the code would crash if any of the casings in the "casings" parameter are null.
If we want to prevent against that we should have a check when the data is passed to the component I think, to prevent rechecking for nulldata on every render.